Yearn-agents/debt-strategies/src/agent.ts

```diff
@@ -0,0 +1,80 @@
+import {  
+  Finding, 
+  TransactionEvent, 
+  getJsonRpcUrl,
+  FindingSeverity,
+  FindingType
+} from 'forta-agent'
+import Web3 from 'web3'
+import MakerFetcher from "./maker.fetcher"
+import {FindingGenerator, provideFunctionCallsDetectorHandler} from 'forta-agent-tools'
+import {tendABI} from "./abi"
+
+const web3: Web3 = new Web3(getJsonRpcUrl())
+const fetcher: MakerFetcher = new MakerFetcher(web3)
+
+const createFindingTendCall: FindingGenerator = (metadata?: ({[key: string]: any}) | undefined) => {
+  return Finding.fromObject({
+    name: "MAKER Strategy - Keeper called tend",
+    description: "A Maker strategy is called by its keeper",
+    severity: FindingSeverity.Info, 
+    type: FindingType.Info, 
+    alertId: "Yearn-2-2",
+    protocol: "Yearn", 
+    metadata: {
+      strategy: metadata?.to
+    }
+  })
+}
+
+export const provideHandleTransaction = ( fetcher: MakerFetcher) => {
+  return async (txEvent: TransactionEvent) => {
+    let findings: Finding[] = [];
+
+    if (!txEvent.status) return findings;
+
+    const activeMakersInfo = await fetcher.getActiveMakersInfo(txEvent.blockNumber);
+    
+    if (activeMakersInfo)
+    {
+      for (let maker of activeMakersInfo)
+      {        
+        if (!maker.checkedRatio) {
+          if (maker.currentMakerVaultRatio.comparedTo(maker.collateralizationRatio.minus(maker.rebalanceTolerance)) < 0){
+            findings.push(
+              Finding.fromObject({
+                name: "MAKER Strategy - MakerVaultRatio is less than acceptable ratio",
+                description: "strategy.getCurrentMakerVaultRatio() < strategy.collateralizationRatio() - strategy.rebalanceTolerance()",
+                severity: FindingSeverity.Medium, 
+                type: FindingType.Info, 
+                alertId: "Yearn-2-1",
+                protocol: "Yearn", 
+                metadata: {
+                  strategy: maker.address
+                }
+              })
+            )
+          }
+          maker.checkedRatio = true
```

Comment:
Is this needed because you can have the same strategy in different vaults? If yes, wouldn't the findings related to calling `tend` be repeated? If no, why do you need this `checkedRatio`?

Reply:
checked Ratio is used here to avoid repeating emitting the finding related to MakerVaultRatio for every tx in the same block.
The reason behind is that ratios are fetched based on the state of a block. If strategy.getCurrentMakerVaultRatio() < strategy.collateralizationRatio() - strategy.rebalanceTolerance() is true for a block, it will be true for every tx in the block.